apps/dashboard/src/main/java/com/akto/action/DashboardAction.java

```diff
@@ -72,39 +77,92 @@ public String fetchRiskScoreCountMap(){
-                Filters.nin("_id.apiInfoKey.apiCollectionId", demoCollections),
-                Filters.eq(TestingRunIssues.TEST_RUN_ISSUES_STATUS,  GlobalEnums.TestRunIssueStatus.OPEN))       
-            );
+        Bson openIssuesMatchStage = Aggregates.match(Filters.and(
+                filters,
+                Filters.in(TestingRunIssues.TEST_RUN_ISSUES_STATUS, GlobalEnums.TestRunIssueStatus.OPEN.name())
+        ));
 
-        // issues that have been created till start timestamp
-        oldOpenCount = TestingRunIssuesDao.instance.count(
-                Filters.and(
-                        Filters.nin("_id.apiInfoKey.apiCollectionId", demoCollections),
-                        Filters.lte(TestingRunIssues.CREATION_TIME, startTimeStamp),
-                        Filters.ne(TestingRunIssues.TEST_RUN_ISSUES_STATUS,  GlobalEnums.TestRunIssueStatus.IGNORED)
-                )
+        pipeline.add(openIssuesMatchStage);
+        totalIssuesCountDayWise = new ArrayList<>();
+
+        GroupByTimeRange.groupByAllRange(daysBetween, pipeline, TestingRunIssues.CREATION_TIME, "totalIssues");
+        MongoCursor<BasicDBObject> cursor = TestingRunIssuesDao.instance.getMCollection().aggregate(pipeline, BasicDBObject.class).cursor();
+        while (cursor.hasNext()) {
+            BasicDBObject document = cursor.next();
+            if(document.isEmpty()) continue;
+            totalIssuesCountDayWise.add(document.getInt("totalIssues"));
+        }
+        cursor.close();
+
+        return SUCCESS.toUpperCase();
+    }
+
+    List<HistoricalData> historicalData;
+    public String fetchAllHistoricalData() {
+        long daysBetween = (endTimeStamp - startTimeStamp) / ONE_DAY_TIMESTAMP;
+
+        List<Bson> pipeline = new ArrayList<>();
+
+        Bson notIncludedCollections = UsageMetricCalculator.excludeDemosAndDeactivated(HistoricalData.API_COLLECTION_ID);
+        Bson filter = Filters.and(
+                notIncludedCollections,
+                Filters.gte("time", startTimeStamp),
+                Filters.lte("time", endTimeStamp)
         );
+        pipeline.add(Aggregates.match(filter));
+
+        historicalData = new ArrayList<>();
+
+        if(daysBetween > 30 && daysBetween <= 210) {
+            addGroupAndProjectStages(pipeline, "week");
+        } else if(daysBetween > 210) {
+            addGroupAndProjectStages(pipeline, "month");
+        }
+
+        MongoCursor<HistoricalData> cursor = HistoricalDataDao.instance.getMCollection().aggregate(pipeline, HistoricalData.class).cursor();
+        while(cursor.hasNext()) {
+            historicalData.add(cursor.next());
+        }
+        cursor.close();
 
         return SUCCESS.toUpperCase();
     }
 
+    private void addGroupAndProjectStages(List<Bson> pipeline, String dateUnit) {
+        Bson groupStage = Aggregates.group(
+                new Document(dateUnit, new Document("$" + dateUnit, new Document("$toDate", new Document("$multiply", Arrays.asList("$time", 1000))))),
```

Comment:
these look like complex operations, please check compatibitly with the other dbs, we support. Take list from here: https://github.com/akto-api-security/akto/blob/master/libs/dao/src/main/java/com/akto/util/DbMode.java#L12 . Check for all the operations in this function

Reply:
yes all the other dbs we use support these aggregation stages